app/src/main/java/io/appium/uiautomator2/utils/AXWindowHelpers.java

```diff
@@ -87,7 +89,18 @@ private static AccessibilityNodeInfo getActiveWindowRoot() {
     }
 
     private static List<AccessibilityWindowInfo> getWindows() {
-        return CustomUiDevice.getInstance().getUiAutomation().getWindows();
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
+            return CustomUiDevice.getInstance().getUiAutomation().getWindows();
+        }
+
+        int currentDisplayId = UiAutomatorBridge.getInstance().getCurrentDisplayId();
+
+        SparseArray<List<AccessibilityWindowInfo>> windowsOnAllDisplays= CustomUiDevice
+                .getInstance()
+                .getUiAutomation()
+                .getWindowsOnAllDisplays();
+
+        return windowsOnAllDisplays.get(currentDisplayId);
```

Comment:
might it be that windowsOnAllDisplays does not contain the given display identifier?